pandas/tests/util/test_show_versions.py

```diff
@@ -26,11 +33,47 @@
     "ignore:Distutils:UserWarning"
 )
 @pytest.mark.filterwarnings("ignore:Setuptools is replacing distutils:UserWarning")
-def test_show_versions(capsys):
+def test_show_versions(tmpdir):
+    # GH39701
+    as_json = os.path.join(tmpdir, "test_output.json")
+
+    pd.show_versions(as_json=as_json)
+
+    with open(as_json) as fd:
+        # check if file output is valid JSON, will raise an exception if not
+        result = json.load(fd)
+
+    # Basic check that each version element is found in output
+    expected = {
+        "system": _get_sys_info(),
+        "dependencies": _get_dependency_info(),
+    }
+
+    assert result == expected
+
+
+def test_show_versions_console_json(capsys):
+    # GH39701
+    pd.show_versions(as_json=True)
+    stdout = capsys.readouterr().out
+
+    # check valid json is printed to the console if as_json is True
+    result = json.loads(stdout)
+
+    # Basic check that each version element is found in output
+    expected = {
+        "system": _get_sys_info(),
+        "dependencies": _get_dependency_info(),
+    }
+
+    assert result == expected
+
+
+def test_show_versions_console(capsys):
     # gh-32041
-    pd.show_versions()
-    captured = capsys.readouterr()
-    result = captured.out
+    # GH39701
```

Comment:
I should've picked up on this earlier, but this one didn't need changing, as it was in the existing test
```suggestion
    # gh-32041
```